### PR TITLE
First pass on the addon and some tests.

### PR DESCRIPTION
**Pull request by patrickberkeley**

### CODE REVIEW

**Comment by alanpeabody** [comment]
:+1:

File: addon/components/input-mask.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    });
+
+    // The input mask changes the value of the input from the original to a
+    // formatted version. We need to manually send that change back to the
+    // controller.
+    this.set('value', this.$().val());
+  }.on('didInsertElement'),
+
+  removeMask: function() {
+    this.$().inputmask('remove');
```
Comment (beerlington) [inline addon/components/input-mask.js:24]: Does this not happen automatically when the component is destroyed?

Comment (patrickberkeley) [inline addon/components/input-mask.js:24]: I'm not entirely positive. But if I don't have this I see the same number of event listeners before _and_ after the view is destroyed.

To test I created an app that uses this input in a view and toggle a conditional with the 'Remove' button that takes the input out of the DOM.

### with inputmask('remove')

screen shot 2015-01-25 at 3 14 33 pm

### without inputmask('remove')

screen shot 2015-01-25 at 3 12 39 pm

Without unbinding the inputmask plugin, the number of listeners increases with every insert-destroy cycle. When it is unbound, the listeners go up and down depending on whether or not the input is in the DOM.

Comment (beerlington) [inline addon/components/input-mask.js:24]: Interesting, looks like I need to update my post :)

**Pull request merged**